src/app/shared/pricing-definitions/pricing-definitions-table-data-source.ts

```diff
@@ -0,0 +1,232 @@
+        headerClass: 'd-none d-xl-table-cell col-30p',
+        class: 'd-none d-xl-table-cell col-30p',
+        sortable: true,
+      },
+      {
+        id: 'staticRestrictions.validTo',
+        name: 'settings.pricing.valid_to',
+        formatter: (validTo: Date) => this.datePipe.transform(validTo),
+        headerClass: 'col-15p',
+        class: 'd-none d-xl-table-cell col-30p',
+        sortable: true,
+      },
+      {
+        id: 'dimensions.flatFee.price',
+        name: 'settings.pricing.flat_fee',
+        formatter: (price: number) => {
+          if (price === undefined) {
+            return '-';
+          }
+          return this.appPricingDimensionsUnit.transform('flat_fee_formatted_price', price);
+        },
+        headerClass: 'col-15p',
+        class: 'col-15p',
+      },
+      {
+        id: 'dimensions.energy.price',
+        name: 'settings.pricing.energy',
+        formatter: (price: number) => {
+          if (price === undefined) {
+            return '-';
+          }
+          return this.appPricingDimensionsUnit.transform('energy_formatted_price', price);
+        },
+        headerClass: 'col-15p',
+        class: 'col-15p',
+      },
+      {
+        id: 'dimensions.chargingTime.price',
+        name: 'settings.pricing.charging_time',
+        formatter: (price: number) => {
+          if (price === undefined) {
+            return '-';
+          }
+          return this.appPricingDimensionsUnit.transform('charging_time_formatted_price', price);
+        },
+        headerClass: 'col-15p',
+        class: 'col-15p',
+      },
+      {
+        id: 'dimensions.parkingTime.price',
+        name: 'settings.pricing.parking_time',
+        formatter: (price: number) => {
+          if (price === undefined) {
+            return '-';
+          }
+          return this.appPricingDimensionsUnit.transform('parking_time_formatted_price', price);
+        },
+        headerClass: 'col-15p',
+        class: 'col-15p',
+      },
+    ];
+    return columns;
+  }
+
+  public buildTableActionsDef(): TableActionDef[] {
+    const tableActionsDef = super.buildTableActionsDef();
+    if (this.canCreatePricingDefinition) {
+      tableActionsDef.unshift(this.createAction);
+    }
+    return tableActionsDef;
+  }
+
+  public buildTableDynamicRowActions(pricingDefinition: PricingDefinition): TableActionDef[] {
+    const rowActions: TableActionDef[] = [];
+    if (pricingDefinition.canUpdate) {
+      rowActions.push(this.editAction);
+    }
+    if (pricingDefinition.canDelete) {
+      rowActions.push(this.deleteAction);
+    }
+    return rowActions;
+  }
+
+  public actionTriggered(actionDef: TableActionDef) {
+    // Action
+    switch (actionDef.id) {
+      case PricingButtonAction.CREATE_PRICING_DEFINITION:
+        if (actionDef.id) {
+          (actionDef as TableCreatePricingDefinitionActionDef).action(PricingDefinitionDialogComponent,
+            this.dialog,
+            {
```

Comment:
OL like below (choose one and stick to it, OL or ML but not a "salade nicoise" in the same file 😊)

Reply:
will commit change ✅